### ISSUE TITLE
Padding batch data with zeros might prevent model from converging

### ISSUE THREAD

**Issue by isobretatel** [keras-rl/keras-rl#261]
https://github.com/keras-rl/keras-rl/blob/master/rl/memory.py
"Your gradients will likely saturate at a zero very soon."

**Comment by mirraaj** [comment]
Hello @isobretatel ,

What makes you think so?

**Comment by isobretatel** [comment]
https://www.quora.com/Is-padding-batch-data-with-zeroes-when-training-neural-network-harmless?__filter__=&__nsrc__=2&__snid3__=3410482604

**Comment by isobretatel** [comment]
1. The agent should not handle the window. It should be responsibility of the environment.
2. The environment should return the window. 
3. The memory should just store and return the window. The memory should not be aware of what is inside of the window, or even know the size of the window.
